python/dlr/api.py

```diff
@@ -1,344 +1,50 @@
-        """Get the size and the dimenson of the index-th output.
-
-        Parameters
-        __________
-        index : int
-            The index of the output.
-
-        Returns
-        _______
-        size : int
-            The size of the index-th output.
-        dim : int
-            The dimension of the index-th output.
-        """
-        idx = ctypes.c_int(index)
-        size = ctypes.c_longlong()
-        dim = ctypes.c_int()
-        _check_call(_LIB.GetDLROutputSizeDim(byref(self.handle), idx,
-                                                      byref(size), byref(dim)))
-        return size.value, dim.value
-
-    def _get_output_shape(self, index):
-        """Get the shape for the index-th output.
-
-        Parameters
-        __________
-        index : int
-            The index of the output.
-
-        Returns
-        _______
-        shape : list
-            The shape of the index-th output.
-        """
-        size, dim = self._get_output_size_dim(index)
-        if not self.output_size_dim:
-            self.output_size_dim = [(0, 0)] * self._get_num_outputs()
-        self.output_size_dim[index] = (size, dim)
-        shape = np.zeros(dim, dtype=np.int64)
-        _check_call(_LIB.GetDLROutputShape(byref(self.handle),
-                                                    c_int(index),
-                    shape.ctypes.data_as(ctypes.POINTER(ctypes.c_longlong))))
-        return shape
-
-    def _get_output(self, index):
-        """Get the index-th output
-
-        Parameters
-        __________
-        index : int
-            The index of the output.
+    @abc.abstractmethod
+    def get_input(self, name, shape=None):
+        raise NotImplementedError
 
-        Returns
-        _______
-        out : np.array
-            A numpy array contains the values of the index-th output
-        """
-        if index >= len(self.output_shapes) or index < 0:
-            raise ValueError("index is expected between 0 and "
-                             "len(output_shapes)-1, but got %d" % index)
+    @abc.abstractmethod
+    def run(self, input_data):
+        raise NotImplementedError
 
-        output = np.zeros(self.output_size_dim[index][0], dtype=np.float32)
-        _check_call(_LIB.GetDLROutput(byref(self.handle), c_int(index),
-                    output.ctypes.data_as(ctypes.POINTER(ctypes.c_float))))
-        out = output.reshape(self.output_shapes[index])
-        return out
+# Wrapper class
+class DLRModel(IDLRModel):
+    def __init__(self, model_path, dev_type='cpu', dev_id=0):
+        #Determine if 3rdparty package needed
+        if os.path.isfile(model_path):    
+            if model_path.endswith('.pb'):
+                from .tf_model import TFModelImpl
+                self._impl = TFModelImpl(model_path, dev_type, dev_id)
+        else:
+            for (dirpath, dirnames, filenames) in os.walk(model_path):
+                for filename in filenames:
```

Comment:
Can we throw "Can not load the model" error if multiple pb files found

Reply:
The current TVM logic in dlr.cc uses the first set of model artifacts found under directory (assumption is that the directory is either decompressed from tarball, or prepared in the same folder structure as the tarball). Best to keep the behaviors consistent.